### DISCUSSION TITLE
Allow node_modules to be considered by JsConfigPathsPlugin

### DISCUSSION THREAD

**Discussion by fabien** [vercel/next.js#27241] [Ideas]
### Describe the feature you'd like to request

Currently, as an optimization, any path with `node_modules` will be skipped by the `JsConfigPathsPlugin` which is responsible for resolving paths/aliases from `jsconfig.json` and `tsconfig.json`.

However, when using `next-transpile-modules` it should be possible to tap into those aliases from within `node_modules` (especially in a monorepo setup, see [this issue](https://github.com/martpie/next-transpile-modules/issues/154#issuecomment-881491414) which was hard to fix because it was being masked in a very particular way by the symlinking).

At the moment, the `node_modules` regex is hardcoded, but I would like to be able to configure the regex (or even function) to determine which paths to skip:

https://github.com/vercel/next.js/blob/dd029f559fb014ce5ecdd06155ed15368ce39fc6/packages/next/build/webpack/plugins/jsconfig-paths-plugin.ts#L168

### Describe the solution you'd like

I propose a `next.config.js` option (perhaps, experimental at first) to set this regex or supply a function to filter out those paths.

### Describe alternatives you've considered

I'm currently forced to perform a bypass, using a custom `JsConfigPathsPlugin` (98% identical) that allows `node_modules`. It replaces the default instance at runtime.

**Comment by smnh** [comment]
I've also tried to use custom Webpack resolve plugin, but it doesn't work in SSR mode when I try to replace internal module required within node_modules. Here is a related issue: https://github.com/vercel/next.js/issues/28500

@fabien in your setup with a custom `JsConfigPathsPlugin` have you managed to make the SSR to pick up the replaced module?
